nifi-nar-bundles/nifi-solr-bundle/nifi-solr-processors/src/main/java/org/apache/nifi/processors/solr/SolrProcessor.java

```diff
@@ -120,6 +202,22 @@ protected final SolrClient getSolrClient() {
                         .explanation("A collection must specified for Solr Type of Cloud")
                         .build());
             }
+
+            if (!context.getProperty(ZK_CLIENT_TIMEOUT).isSet()) {
+                problems.add(new ValidationResult.Builder()
+                        .subject(ZK_CLIENT_TIMEOUT.getName())
+                        .valid(false)
+                        .explanation("A ZooKeeper Client Timeout must specified for Solr Type of Cloud")
+                        .build());
+            }
+
+            if (!context.getProperty(ZK_CONNECTION_TIMEOUT).isSet()) {
```

Comment:
This is 'dead code' as isSet() will always return 'true', since there is a defaultValue